storage/grpc_client.go

```diff
@@ -625,7 +625,51 @@ func (c *grpcStorageClient) UpdateObjectACL(ctx context.Context, bucket, object
 // Media operations.
 
 func (c *grpcStorageClient) ComposeObject(ctx context.Context, req *composeObjectRequest, opts ...storageOption) (*ObjectAttrs, error) {
-	return nil, errMethodNotSupported
+	s := callSettings(c.settings, opts...)
+	if s.userProject != "" {
+		ctx = setUserProjectMetadata(ctx, s.userProject)
+	}
+
+	dstObjPb := req.dstObject.attrs.toProtoObject(req.dstBucket)
+	dstObjPb.Name = req.dstObject.name
+	if err := applyCondsProto("ComposeObject destination", -1, req.dstObject.conds, dstObjPb); err != nil {
+		return nil, err
+	}
+	if req.sendCRC32C {
+		dstObjPb.Checksums.Crc32C = &req.dstObject.attrs.CRC32C
+	}
+
+	srcs := []*storagepb.ComposeObjectRequest_SourceObject{}
+	for _, src := range req.srcs {
+		srcObjPb := &storagepb.ComposeObjectRequest_SourceObject{Name: src.name}
+		srcObjPb.Generation = src.gen // what if default value?
```

Comment:
Is this question about "What if the src object generation is actually just the default value (e.g. `0`)"?

Reply:
If gen = 0 it might have unintended behavior because gen = 0 usually represents object doesn't exist.

Reply:
I think it's handled here: https://github.com/googleapis/google-cloud-go/blob/0d124aea0fd34510d58167a2b64a17c70c07e43e/storage/storage.go#L2123